### ISSUE TITLE
.npy files

### ISSUE THREAD

**Issue by pjrockzzz** [berkgulay/WeatherPredictionFromImage#29]
from where, we can get .npy data sets ?

If not, how to apply images data set to given code ?

**Comment by berkgulay** [comment]
Hello Panchajanya,

Below you can find our dataset links for each class of our work. Inside our codes there is a function (**_image_to_matrix_**) to create .npy files and respective labels. You can find that function in "tools.py" file. We also mentioned our project's "Final Report" in Github. We hope, you can make use of our works. If you could give proper credits to our work and star our project in Github, we would be pleased.

https://drive.google.com/open?id=1j9nLFoAA5xxC5DplQd-mgbysVsMcJEcz -- (train images for class-0)
https://drive.google.com/open?id=1KO0ryOH6j4pFYTJpSyGxL1iH1f84dNU7 -- (train images for class-1)
https://drive.google.com/open?id=1MWLGbv82_aEZo3h84pQAMRHkvCvsWSkA -- (train images for class-2,3,4)

0 --> Cloudy
1 --> Sunny
2 --> Rainy
3 --> Snowy
4 --> Foggy

Have a nice day,
Berk Gülay
Hacettepe University Computer Science

**Comment by pjrockzzz** [comment]
Thankyou so much @berkgulay

**Comment by Allesgute1** [comment]
Hello, I have download your pictures and your code,
How can I run this code?
I haven't found a space to use the function "image_to_matrix"?
Looking forward to your reply. Thank you

**Comment by berkgulay** [comment]
Hey,
You can run our codes in your favorite Python IDE using a python interpreter. (just open source codes in IDE)
"image_to_matrix" function provides you numpy files to train your Neural Network model. By this function, you can convert downloaded images (with their labels) to .npy files. Labels will be given to you in .npy files using "folder_name" of that image (so, as image class "folder_name" will be used).

If you have further questions, I am here :) Hope you can make use of our works.

**Comment by Allesgute1** [comment]
Thank for your reply!
 I want to use the method:RF and CNN, I think that the image_to_matrix function was used to create the .npy files for CNN and create_features function was used to create the .npy filse for RF and other machine learning method, have I understood correctly?
  

 

 ------------------ 原始邮件 ------------------
  发件人: "Berk"<notifications@github.com>;
 发送时间: 2018年11月21日(星期三) 凌晨5:00
 收件人: "berkgulay/WeatherPredictionFromImage"<WeatherPredictionFromImage@noreply.github.com>;
 抄送: "张思佳"<752729797@qq.com>; "Comment"<comment@noreply.github.com>; 
 主题: Re: [berkgulay/WeatherPredictionFromImage] .npy files (#29)

 

 
Hey,
You can run our codes in your favorite Python IDE using a python interpreter. (just open source codes in IDE)
"image_to_matrix" function provides you numpy files to train your Neural Network model. By this function, you can convert downloaded images (with their labels) to .npy files. Labels will be given to you in .npy files using "folder_name" of that image (so, as image class "folder_name" will be used).
 
If you have further questions, I am here :) Hope you can make use of our works.

**Comment by Allesgute1** [comment]
Thank for your reply!    I want to use the method:RF and CNN, I think that the image_to_matrix function was used to create the .npy files for CNN and create_features function was used to create the .npy filse for RF and other machine learning method, have I understood correctly?
   And I also have a question, I have so plenty of .npy files, How can I use them?
   Looking forward to your reply. Thank you!

  

 

 ------------------ 原始邮件 ------------------
  发件人: "Berk"<notifications@github.com>;
 发送时间: 2018年11月21日(星期三) 凌晨5:00
 收件人: "berkgulay/WeatherPredictionFromImage"<WeatherPredictionFromImage@noreply.github.com>;
 抄送: "张思佳"<752729797@qq.com>; "Comment"<comment@noreply.github.com>; 
 主题: Re: [berkgulay/WeatherPredictionFromImage] .npy files (#29)

 

 
Hey,
You can run our codes in your favorite Python IDE using a python interpreter. (just open source codes in IDE)
"image_to_matrix" function provides you numpy files to train your Neural Network model. By this function, you can convert downloaded images (with their labels) to .npy files. Labels will be given to you in .npy files using "folder_name" of that image (so, as image class "folder_name" will be used).
 
If you have further questions, I am here :) Hope you can make use of our works.

**Comment by berkgulay** [comment]
Yes your general understanding is correct. **_We used created numpy files by "image_to_matrix" for training procedures of CNN algorithm. And also we extracted features (using our Features.py functions) from standart rgb images by "ImageDescriptor.py" file's "describe" function. This "describe" function also creates numpy files for training of RF, Decision Tree or SVM algorithms._** After that point you should use our algorithm specific files like "RF_train" , "CNN_train" , "DT_train" etc. to train your own models. You can change parameters of these algorithms before training. Maybe you can create a model better than ours. Up to you. Rest is your job and knowledge :) If you have not experienced about ML field, learn/read a bit about ML and try basic tutorials or easier applications first. If you have some experience, then good luck!

**Comment by berkgulay** [comment]
You can also **fork** the project and work on your own instance ;)

**Comment by Allesgute1** [comment]
Thank you so much！

**Comment by vineetjai** [comment]
@berkgulay I had runned `image_to_matrix` function from `tools.py` but after certain images processed my RAM(8GB) got full and further images is not processed. I have 3 questions:-
1. Can you please tell me your system specification?
2. Did you run it on cloud or on local computer?
3. Also, can you further optimise your code so less memory will be taken because it will not be able to run for large dataset if it stores everything on memory?

**Comment by vineetjai** [comment]
@berkgulay I had tried to reduce batch size to overcome memory storage problem but it is creating multiple `.npy` files. Can I merge them into single file for training?

**Comment by berkgulay** [comment]
Hey vineetjai, 

1) My memory was 16 GB and our image dataset size was over 120k.
2) We had run methods in our local machines and got no error. But you can use cloud services like "Colab" --> https://colab.research.google.com/ and this can be a fast solution for you.
3) You can try to manage batches more feasibly. And then you can merge _.npy_ files. But you should also merge their corresponding label files in another big npy. 

3) I will also check if I can optimize method..

*I am closing this issue. If you have further questions or problems please open new issue. Thanks!
